### PR TITLE
Consolidations in work form

### PR DESCRIPTION
**Pull request by actlikewill**
- This adds consolidations to the work form

https://www.loom.com/share/50b88060883d4e9ba321c9bd1c0d13af?sid=d854a155-5c52-49b1-b15b-d9220d157307

closes laws-africa/indigo-lawsafrica#1188

### CODE REVIEW

File: indigo_app/views/works.py [hunk 1]
```diff
@@ -1535,3 +1536,56 @@ def get_context_data(self, **kwargs):
...
+                date = datetime.datetime.today()
+                # if (
+                #     self.place.place_settings.is_consolidation and
+                #     self.place.place_settings.as_at_date and
+                #     self.work.principal and
+                #     self.work.publication_date and
+                #     self.work.publication_date > self.place.place_settings.as_at_date
+                # ):
+                #
+                #     date = self.place.place_settings.as_at_date
```
Comment (actlikewill) [inline indigo_app/views/works.py:1577]: @goose-life How do I determine the correct place settings to use for a place? it seems we  do not have a one to one relation for a place and place settings

File: indigo_app/templates/indigo_api/_work_form_content.html [hunk 2]
```diff
@@ -454,6 +464,18 @@ <h4 class="text-muted work-frbr-uri">{{ work.frbr_uri }}</h4>
...
+        <div class="card mb-3">
+          <div class="card-body">
+            <div id="work-form-repeals-made">
+              {% include 'indigo_api/_work_form_consolidations.html' %}
+            </div>
+          </div>
+        </div>
+      {% endif %}
+    {% endblock work-form-consolidations %}
+
```
Comment (goose-life) [inline indigo_app/templates/indigo_api/_work_form_content.html:478]: I would just move this up to be above the first Amendment block

File: indigo_app/views/works.py [hunk 3]
```diff
@@ -1535,3 +1536,58 @@ def get_context_data(self, **kwargs):
...
+                    "DELETE": form.cleaned_data["DELETE"],
+                })
+
+            if self.request.POST.get("consolidation") == "add":
+                date = ""
+                if (
+                    self.work.principal and
+                    self.work.publication_date and
+                    self.place.settings.is_consolidation and
+                    self.place.settings.as_at_date and
```
Comment (goose-life) [inline indigo_app/views/works.py:1574]: rather use self.work.as_at_date() as that takes the optional as_at_date_override on a work into account

File: indigo_app/templates/indigo_api/_work_form_content.html (hunk 2 above)
Comment (goose-life) [reply indigo_app/templates/indigo_api/_work_form_content.html:478]: ('Amended by')

Comment (actlikewill) [reply indigo_app/templates/indigo_api/_work_form_content.html]: @goose-life updated

**Review by goose-life — APPROVED** [review]

**Pull request merged**